WordPress/Classes/Services/AccountService.m

```diff
@@ -242,6 +258,39 @@ - (NSUInteger)numberOfAccounts
     return count;
 }
 
+/**
+ Returns all accounts currently existing in core data.
+
+ @return An array of WPAccounts.
+ */
+- (NSArray<WPAccount *> *)allAccounts
+{
+    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:@"Account"];
+    NSError *error = nil;
+    NSArray *fetchedObjects = [self.managedObjectContext executeFetchRequest:fetchRequest error:&error];
+    if (error) {
+        return @[];
+    }
+    return fetchedObjects;
+}
+
+/**
+ Checks an account to see if it is just used to connect to Jetpack.
+
+ @param account The account to inspect.
+ @return True if used only for a Jetpack connection.
+ */
+- (BOOL)isJetpackAccount:(WPAccount *)account
+{
+    // If an account has multiple blogs, or zero blogs, then it is not an account
+    // that was connected via Jetpack login while another default wpcom account existed.
```

Comment:
I don't think that's accurate. With the logging changes in #12411 I tested:

1. Installed the app and logged in as kokejb7
2. Added self-hosted sandbox.koke.me, connected to `kokeme`. Logged in from stats
3. Added self-hosted nosenada.es, connected to `kokeme`. Stats worked right away without a second login
4. Signed out from Me > Sign out

On relaunch, this is the current state:

```
All accounts and blogs:
<Account username: kokeme ID: 442746, Email: koke@amedias.org (verified) ObjectID: x-coredata://26A6E50E-DCDD-4053-A67F-FA7E48A92DB1/Account/p6>
└─ <Blog Name: Koke's .org Sandbox URL: https://sandbox.koke.me XML-RPC: https://sandbox.koke.me/xmlrpc.php wp.com account: kokeme blogId: 137551970 plan: Free (2002) ObjectID: x-coredata://26A6E50E-DCDD-4053-A67F-FA7E48A92DB1/Blog/p2176>
└─ <Blog Name: Sólo sé que no sé nada URL: https://nosenada.es XML-RPC: https://nosenada.es/xmlrpc.php wp.com account: kokeme blogId: 94917218 plan: Free (2002) ObjectID: x-coredata://26A6E50E-DCDD-4053-A67F-FA7E48A92DB1/Blog/p2177>
```

I think that we should just see if an account has any WordPress.com blogs and promote that to default.

Reply:
Oh good catch there. 😬 
> I think that we should just see if an account has any WordPress.com blogs and promote that to default.

There's still a situation where a user might have had a default account with 0 blogs.  Lemme tweak the logic a bit. I like the idea of checking for wpcom hosted blogs, and we can add an extra check for 0 blogs case.